typing/btype.mli

```diff
@@ -171,6 +171,9 @@ val not_marked_node: type_expr -> bool
         (* Return true if a type node is not yet marked *)
 val flip_mark_node: type_expr -> unit
         (* Mark a type node. No [repr]'ing *)
+val logged_mark_node: type_expr -> unit
+        (* Mark a type node, logging the marking so it can be backtracked.
+           No [repr]'ing *)
```

Comment:
If I understand correctly, the other `*_mark_node` functions are designed to be followed by an `unmark_*` call for cleanup, while this function is designed to be followed by a `backtrack` call for cleanup. The `unmark_*` approach is more efficient, but it assumes that the function has a simple control flow (in particular, no exception failures in-between), the `backtrack` approach is more general.

Some comments:
- this could possibly be explained in comments
- this suggests an `.mli` organization with first `not_marked_node`, which is unmarking-agnostic, and then two groups of functions depending on whether they use unmarking or backtracking
- the `try_` variant would be meaningful in either cases, so it would make sense to expose `try_logged_mark_mode`, and in fact wouldn't it be useful in `check_scope_escape`, if you define `orig_level` first?

Reply:
I roughly implemented your suggestion:
- `check_scope_escape` now uses a brand new `try_logged_mark_node`. While this is a tiny patch, it isn't reducible to a simple search&replace as the previous iteration of this PR, so I'd encourage you to read it and not trust me blindly :)
- I've regrouped the `.mli` declarations as suggested (I think)
- I've added some comments regarding the (absence of) logging and its implications. Probably not as clear or detailed as you'd like, but I'm stopping here nonetheless. I'll be happy to review your documentation changes on that particular interface if you wish to make some.

Reply:
Okay with the intermediate amount of documentation, thanks.